src/ua/griddynamics/CoordinateChecker.java

```diff
@@ -0,0 +1,47 @@
+package ua.griddynamics;
+
+import java.util.Scanner;
+
+public class CoordinateChecker {
+    private final Scanner scanner = new Scanner(System.in);
+
+    public String[] scanAndCheckUserInput() {
+        String[] userInput;
+        do {
+            System.out.print("Set/delete mines marks (x and y coordinates): > ");
+            userInput = scanner.nextLine().split(" ");
+        } while (userInput.length != 3
+                || !isCoordinateDigit(userInput[0]) && !isCoordinateDigit(userInput[1])
+                || !isCoordinateValid(userInput)
+                || !isMarkValid(userInput));
+        return userInput;
+    }
+
+    private boolean isMarkValid(String[] userInput) {
+        String mark = userInput[2];
+        return mark.equals("free") || mark.equals("mine");
```

Comment:
Don't use text and magic numbers in business logic. Create constants when it is needed

Reply:
Done